app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheet.kt

```diff
@@ -89,7 +91,12 @@ class BloodPressureEntrySheet : BottomSheetActivity() {
   private fun sheetCreates(): Observable<UiEvent> {
     val patientUuid = intent.extras!!.getSerializable(KEY_UUID) as UUID
     val openAs = intent.extras!!.getSerializable(KEY_OPEN_AS) as OpenAs
-    return Observable.just(BloodPressureEntrySheetCreated(openAs, patientUuid))
+    return Observable
+        .just(BloodPressureEntrySheetCreated(openAs, patientUuid))
+        // This delay stops the race condition (?) that happens frequently with replay().refCount()
+        // in the controller. Temporary workaround until we figure out what exactly is going on.
+        .delay(100L, TimeUnit.MILLISECONDS)
+        .cast()
```

Comment:
The commit name looks wrong…? `Prefill BP when opening BP entry sheet for updates`

Reply:
TIL about `cast()`. Useful!

Reply:
Whoops. Will fix this.

Reply:
Squashed this commit with the actual prefill BP fields commit since this was a workaround to fix a bug in that commit.